File: Queues/queue_linkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

front = None
rear = None

def enqueue(item):
    global front, rear
    new_node = Node(item)

    if rear is None:
        front = rear = new_node
    else:
        rear.next = new_node
        rear = new_node

    print("Inserted", item)

enqueue(10)
enqueue(20)
enqueue(30)

File: Queues/QueueLinkedList.java
class Node
{
    int data;
    Node next;

    Node(int data)
    {
        this.data = data;
        this.next = null;
    }
}

class QueueLinkedList
{
    Node front = null, rear = null;

    void enqueue(int item)
    {
        Node newNode = new Node(item);

        if (rear == null)
        {
            front = rear = newNode;
        }
        else
        {
            rear.next = newNode;
            rear = newNode;
        }
        System.out.println("Inserted " + item);
    }

    public static void main(String[] args)
    {
        QueueLinkedList q = new QueueLinkedList();
        q.enqueue(10);
        q.enqueue(20);
        q.enqueue(30);
    }
}
